http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriberModels.scala

```diff
@@ -0,0 +1,215 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package monitor
+
+import io.circe.syntax._
+import io.circe.{Decoder, DecodingFailure, Encoder}
+import org.broadinstitute.dsde.workbench.google2.JsonCodec.traceIdDecoder
+import org.broadinstitute.dsde.workbench.google2.JsonCodec.traceIdEncoder
+import org.broadinstitute.dsde.workbench.leonardo.JsonCodec._
+import org.broadinstitute.dsde.workbench.leonardo.model.google.{ClusterName, ClusterStatus, IP, OperationName}
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, ClusterProjectAndName, DataprocInfo}
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage._
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.model.google.GcsPath
+
+import scala.util.control.NoStackTrace
+
+sealed trait LeoPubsubMessage {
+  def traceId: Option[TraceId]
+  def messageType: String
+}
+
+object LeoPubsubMessage {
+  final case class StopUpdateMessage(updatedMachineConfig: RuntimeConfig, clusterId: Long, traceId: Option[TraceId])
+      extends LeoPubsubMessage {
+    val messageType = "stopUpdate"
+  }
+
+  case class ClusterTransitionFinishedMessage(clusterFollowupDetails: ClusterFollowupDetails, traceId: Option[TraceId])
+      extends LeoPubsubMessage {
+    val messageType = "transitionFinished"
+  }
+
+  case class CreateCluster(id: Long,
+                            clusterProjectAndName: ClusterProjectAndName,
+                            serviceAccountInfo: ServiceAccountInfo,
+                            dataprocInfo: Option[DataprocInfo],
+                            auditInfo: AuditInfo,
+                            properties: Map[String, String],
+                            jupyterExtensionUri: Option[GcsPath],
+                            jupyterUserScriptUri: Option[UserScriptPath],
+                            jupyterStartUserScriptUri: Option[UserScriptPath],
+                            userJupyterExtensionConfig: Option[UserJupyterExtensionConfig],
+                            defaultClientId: Option[String],
+                            clusterImages: Set[ClusterImage],
+                            scopes: Set[String],
+                            welderEnabled: Boolean,
+                            customClusterEnvironmentVariables: Map[String, String],
+                            runtimeConfig: RuntimeConfig,
+                            traceId: Option[TraceId])
+      extends LeoPubsubMessage {
+    val messageType = "createCluster"
+  }
+
+  final case class ClusterFollowupDetails(clusterId: Long, clusterStatus: ClusterStatus)
+      extends Product
+      with Serializable
+}
+
+final case class PubsubException(message: String) extends Exception
+
+object LeoPubsubCodec {
```

Comment:
Maybe we should move the Codec to another file. I can see this file getting unwieldy pretty fast

Reply:
I think it's okay to keep the codec in the same file...models for this file alone isn't that big...if it gets bigger, we can easily move it then.